org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/themes/Theme.java

```diff
@@ -1,158 +1,179 @@
-				e.printStackTrace();
-			}
-		}
-		return tokenProvider;
-	}
-
-	@Override
-	public String toCSSStyleSheet() {
-		return super.getResourceContent();
-	}
-
-	@Override
-	public boolean isDark() {
-		return dark;
-	}
-
-	@Override
-	public boolean isDefault() {
-		return isDefault;
-	}
-
-	@Override
-	public void initializeViewerColors(StyledText styledText) {
-		ITokenProvider tokenProvider = getTokenProvider();
-		Color background = tokenProvider.getEditorBackground();
-		if (background != null) {
-			styledText.setBackground(background);
-		}
-		Color foreground = tokenProvider.getEditorForeground();
-		if (foreground != null) {
-			styledText.setForeground(foreground);
-		}
-		Color selectionBackground = tokenProvider.getEditorSelectionBackground();
-		if (selectionBackground != null) {
-			styledText.setSelectionBackground(selectionBackground);
-		}
-		Color selectionForeground = tokenProvider.getEditorSelectionForeground();
-		styledText.setSelectionForeground(null);
-		if (selectionForeground != null) {
-			styledText.setSelectionForeground(selectionForeground);
-		}
-	}
-}
+/**
+ *  Copyright (c) 2015-2017 Angelo ZERR.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  Contributors:
+ *  Angelo Zerr <angelo.zerr@gmail.com> - initial API and implementation
+ */
+package org.eclipse.tm4e.ui.themes;
+
+import java.io.InputStream;
+
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.jface.text.rules.IToken;
+import org.eclipse.swt.custom.StyledText;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.tm4e.registry.TMResource;
+import org.eclipse.tm4e.registry.XMLConstants;
+import org.eclipse.tm4e.ui.internal.preferences.PreferenceConstants;
+import org.eclipse.tm4e.ui.themes.css.CSSTokenProvider;
+import org.eclipse.ui.texteditor.AbstractTextEditor;
+
+/**
+ * {@link ITheme} implementation.
+ *
+ */
+public class Theme extends TMResource implements ITheme {
+
+	private static final String DARK_ATTR = "dark";
+	private static final String DEFAULT_ATTR = "default";
+
+	private ITokenProvider tokenProvider;
+
+	private String id;
+	private String name;
+	private boolean dark;
+	private boolean isDefault;
+
+	/**
+	 * Constructor for user preferences (loaded from Json with Gson).
+	 */
+	public Theme() {
+		super();
+	}
+
+	/**
+	 * Constructor for extension point.
+	 * 
+	 * @param element
+	 */
+	public Theme(String id, String path, String name, boolean dark, boolean isDefault) {
+		super(path);
+		this.id = id;
+		this.name = name;
+		this.dark = dark;
+		this.isDefault = isDefault;
+	}
+
+	public Theme(IConfigurationElement ce) {
+		super(ce);
+		id = ce.getAttribute(XMLConstants.ID_ATTR);
+		name = ce.getAttribute(XMLConstants.NAME_ATTR);
+		dark = "true".equals(ce.getAttribute(DARK_ATTR));
```

Comment:
Boolean.parse(...)

Reply:
I pushed last commit with crlf enabled :<
Add correct paring in 360e67f commit.